sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -216,10 +234,10 @@ public async ValueTask WaitForEnvironmentAsync()
                 Task task;
                 lock (s_environmentStateCache)
                 {
-                    if (!s_environmentStateCache.TryGetValue(GetType(), out task))
```

Comment:
GetType is free

Reply:
don't kid yourself - nothing in this life is free